byterun/caml/compatibility.h

```diff
@@ -18,11 +18,24 @@
 #ifndef CAML_COMPATIBILITY_H
 #define CAML_COMPATIBILITY_H
 
+/* minimum API version compatibility: default OCaml 1.0 */
+#ifndef CAML_API_VERSION
+#define CAML_API_VERSION 100
+#endif
+
+#if CAML_API_VERSION >= 400 && !defined(CAML_NAME_SPACE)
+#define CAML_NAME_SPACE
+#endif
+
+#ifndef CAML_NAME_SPACE
+
+#if CAML_API_VERSION < 403
```

Comment:
This is fishy: the test is redundant because lines 26--30 ensure that `CAML_API_VERSION < 400`.

Reply:
p1. I'm rebasing onto trunk now.
p2. I'm moving the `!defined(CAML_NAME_SPACE)` predicate back where it was, because I think it's not meant to apply in cases like this.